### PR TITLE
Delay array expression solving until constant evaluator

### LINKED ISSUE #1570: PIL Analyzing fails if degree is set to `None`
Noticed this issue when prototyping dynamic VadCop (#1558): If we don't set a namespace degree, analyzing the PIL fails in some cases. In particular, we unwrap [here](https://github.com/powdr-labs/powdr/blob/fbf2b4074d0313b05107ff1532a656e467514668/pil-analyzer/src/statement_processor.rs#L533) because [`RepeatedArray`](https://github.com/powdr-labs/powdr/blob/main/ast/src/analyzed/mod.rs#L621) needs to know it. `RepeatedArray` is used to represent statements like `col fixed foo = [1, 2, 3]*`.

### PR DESCRIPTION
**Pull request by Schaeff**
Equivalent to #1623  but in Rust

Closes #1570

@georgwiese I wanted to add a test but everything I try either panics (we sometimes assume the fixed columns to be available in a single size) or runs forever (when I add fixed columns to a machine, I assume witgen keeps increasing the degree and never stops? or it picks the largest size and just takes time?) any thoughts?

### CODE REVIEW

File: ast/src/analyzed/mod.rs [hunk 1]
```diff
@@ -592,9 +593,7 @@ impl Children<Expression> for FunctionValueDefinition {
             FunctionValueDefinition::Expression(TypedExpression { e, type_scheme: _ }) => {
                 Box::new(iter::once(e))
             }
-            FunctionValueDefinition::Array(array) => {
-                Box::new(array.iter().flat_map(|i| i.children()))
-            }
+            FunctionValueDefinition::Array(e) => Box::new(e.children()),
```
Comment (chriseth) [inline ast/src/analyzed/mod.rs:596]:
```suggestion
            FunctionValueDefinition::Array(e) => e.children(),
```

File: ast/src/analyzed/mod.rs [hunk 2]
```diff
@@ -609,9 +608,7 @@ impl Children<Expression> for FunctionValueDefinition {
             FunctionValueDefinition::Expression(TypedExpression { e, type_scheme: _ }) => {
                 Box::new(iter::once(e))
             }
-            FunctionValueDefinition::Array(array) => {
-                Box::new(array.iter_mut().flat_map(|i| i.children_mut()))
-            }
+            FunctionValueDefinition::Array(e) => Box::new(e.children_mut()),
```
Comment (chriseth) [inline ast/src/analyzed/mod.rs:611]:
```suggestion
            FunctionValueDefinition::Array(e) => e.children_mut(),
```

File: executor/src/constant_evaluator/mod.rs [hunk 3]
```diff
@@ -109,8 +109,9 @@ fn generate_values<T: FieldElement>(
         }
         FunctionValueDefinition::Array(values) => {
             assert!(index.is_none());
+            let size_of_repeated_term = values.solve(degree);
```
Comment (chriseth) [inline executor/src/constant_evaluator/mod.rs:112]: Would it make sense to change `solve` to be private?

**Comment by leonardoalt** [comment]
can we merge?

**Comment by chriseth** [comment]
Is the test still outstanding?

**Comment by Schaeff** [comment]
CI is green

**Review by chriseth — APPROVED** [review]

**Pull request merged**